fix: first question has always yes answer

kalloc opened the pull request:
because first bits are usually zero, value is omitted and the answer for the first question is always yes

kalloc committed:
fix: first question has always yes answer

because first bits are usually zero, value is omitted and the answer for the first question is always yes

enovikov11 commented:
Thanks for sending PR. Because it will break existing links, lets check probability of 1 or 0 in start and end.

Seems that hash is crypto strong and distribution will be 50% to 50%, but I'm going to check it and comment it later

kalloc commented:
```js
for(var blockId of [...Array(10).keys()].map(i => i + 14146200)) { 
    provider.getBlock(blockId).then(function({ hash }) { 
        let answers = BigInt(hash).toString(2).split('');
        console.error(blockId, answers[0])
   });
}
```

```output
Promise {<pending>}
VM2390:1 14146209 '1'
VM2390:1 14146209 '1'
VM2390:1 14146209 '1'
VM2390:1 14146209 '1'
VM2390:1 14146209 '1'
VM2390:1 14146209 '1'
VM2390:1 14146209 '1'
VM2390:1 14146209 '1'
VM2390:1 14146209 '1'
```

```js
for(var blockId of [...Array(10).keys()].map(i => i + 14146200)) { 
    provider.getBlock(blockId).then(function({ hash }) { 
        let answers = BigInt(hash).toString(2).split('').reverse();
        console.error(blockId, answers[0])
   });
}
```

```output
Promise {<pending>}
VM2536:1 14146209 '1'
VM2536:1 14146209 '1'
VM2536:1 14146209 '1'
VM2536:1 14146209 '1'
VM2536:1 14146209 '0'
VM2536:1 14146209 '1'
VM2536:1 14146209 '0'
VM2536:1 14146209 '1'
VM2536:1 14146209 '0'
VM2536:1 14146209 '1'
```

```js
for(var blockId of [...Array(10).keys()].map(i => i + 14146200)) { 
    provider.getBlock(blockId).then(function({ hash }) { 
        let answers = BigInt(hash).toString(2).padStart(256, 0).split('');
        console.error(blockId, answers[0])
   });
}
```

```output
Promise {<pending>}
VM2906:4 14146209 '0'
VM2906:4 14146209 '1'
VM2906:4 14146209 '1'
VM2906:4 14146209 '1'
VM2906:4 14146209 '0'
VM2906:4 14146209 '1'
VM2906:4 14146209 '1'
VM2906:4 14146209 '1'
VM2906:4 14146209 '1'
VM2906:4 14146209 '0'
```

enovikov11 commented:
Thanks, bug is confirmed, thats because .toString(2) removes leading zeroes

[...new Set(hashes_14140000_14140099.map(hash => BigInt(hash).toString(2).length))]
Array(8) [ 256, 250, 255, 253, 251, 254, 249, 252 ]
